src/react/hooks.ts

```diff
@@ -176,3 +176,33 @@ export function useInteraction(
 
   return [workbookInteraction, setInteraction];
 }
+
+/**
+ * React hook for returning a triggering callback function for the registered
+ * action trigger
+ * @param {string} id ID of action trigger
+ * @returns {Function} A callback function to trigger the action
+ */
+export function useActionTrigger(id: string) {
+  const client = usePlugin();
+
+  return useCallback(() => {
+    client.config.triggerAction(id);
+  }, [client, id]);
+}
+
+/**
+ * React hook for registering and unregistering an action effect
+ * @param {string} id ID of action effect
+ * @param {Function} effect The function to be called when the action is triggered
+ */
+export function useActionEffect(id: string, effect: Function) {
```

Comment:
Effect should probably be passed through a `useCallbackRef` hook

Reply:
Good catch!